#!/usr/bin/env python3
PACKAGE = "loop_closure"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("hor_res",            int_t,     0, "Horizontal resolution of the scanner",  16,    1,  2048)
gen.add("vert_res",           int_t,     0, "Vertical resolution of the scanner",    16,    1,   128)
gen.add("opening_degree",     int_t,     0, "Vertical opening in degree",            16,    1,    90)
gen.add("step_size",          double_t,  0, "Step size for ray updates",          0.064,  0.0,   1.0)
gen.add("ray_size",           double_t,  0, "Size of each individual ray",         0.01,  0.0,   0.2)

exit(gen.generate(PACKAGE, "lc_test_node", "RayTracer"))